packages/flame/lib/src/components/isometric_tile_map_component.dart

```diff
@@ -8,14 +10,26 @@ import 'position_component.dart';
 /// This is just a pair of <int, int>.
 ///
 /// Represents a position in a matrix, or in this case, on the tilemap.
+@immutable
 class Block {
   /// x and y coordinates on the matrix
-  int x, y;
+  final int x, y;
 
-  Block(this.x, this.y);
+  const Block(this.x, this.y);
 
   @override
   String toString() => '($x, $y)';
+
+  @override
+  bool operator ==(Object other) {
+    if (other is! Block) {
+      return false;
+    }
+    return other.x == x && other.y == y;
+  }
+
+  @override
+  int get hashCode => hashValues(x, y);
```

Comment:
A toVector method would be interesting too, what do you think?

Reply:
Yeah, just using the x, y from the block to create a vector2, we had to that manually on heeve to make the unit movement, so I think could be a good shortcut.